crawl-ref/source/religion.cc

```diff
@@ -752,6 +752,12 @@ void dec_penance(god_type god, int val)
         return;
     }
 
+    if (you.attribute[ATTR_TRAITOR] > 0)
+    {
+        you.attribute[ATTR_TRAITOR]--;
+        you.redraw_title = true;
+    }
+
```

Comment:
Instead of using an attribute for number of gods betrayed, consider having an attribute for which god should grant the traitor title. 
During this step, instead of decrementing you.attribute traitor, check if there are any gods which you have penance for but are not presently worshipping, and set the traitorizor attribute to one of them, likely the one with the most remaining penance. This allows you to implement a logical rule for which title to use when abandoning multiple gods, instead of picking the last god in the god iterator you have penance for.